lib/src/facts/linux/operating_system_resolver.cc

```diff
@@ -21,82 +22,186 @@ namespace bs = boost::system;
 
 namespace facter { namespace facts { namespace linux {
 
-    operating_system_resolver::data operating_system_resolver::collect_data(collection& facts)
+    // Return contents of the os-release file
+    // http://www.freedesktop.org/software/systemd/man/os-release.html
+    static map<string, string> key_value_file(string file)
```

Comment:
I know these release files are generally trivially small, but I'm not a big fan of buffering a file's contents when reading through each line is really what we are doing.  Is there a way the base `os_linux` type could perform this operation and pass the key-value pairs to the derived type after each one is read? That way each derived type can make use of only the keys that they care about too.

Perhaps make this a base member function you can pass a lambda to that gets passed the key and value and then each derived type can process their relevant keys?  Thoughts?